starky/src/get_challenges.rs

```diff
@@ -25,18 +33,24 @@ use crate::proof::*;
 /// or not by the challenger. Observing it here could be redundant in a
 /// multi-STARK system where trace caps would have already been observed
 /// before proving individually each STARK.
-fn get_challenges<F, C, const D: usize>(
+fn get_challenges<F, C, S: Stark<F, D>, const D: usize>(
+    stark: &S,
+    public_inputs: &[F],
     challenger: &mut Challenger<F, C::Hasher>,
     challenges: Option<&GrandProductChallengeSet<F>>,
     trace_cap: Option<&MerkleCap<F, C::Hasher>>,
     auxiliary_polys_cap: Option<&MerkleCap<F, C::Hasher>>,
     quotient_polys_cap: Option<&MerkleCap<F, C::Hasher>>,
+    ctl_vars: Option<&[CtlCheckVars<F, F::Extension, F::Extension, D>]>,
     openings: &StarkOpeningSet<F, D>,
     commit_phase_merkle_caps: &[MerkleCap<F, C::Hasher>],
     final_poly: &PolynomialCoeffs<F::Extension>,
     pow_witness: F,
     config: &StarkConfig,
     degree_bits: usize,
+    num_trace_polys: usize,
+    num_aux_polys: usize,
+    pow_degree: usize,
```

Comment:
`num_trace_polys` and `pow_degree` can be extracted internally from the `stark` parameter no?